### ISSUE TITLE
Wrong reference point when resolving relative object paths in event assignments

### ISSUE THREAD

**Issue by Yuri05** [Open-Systems-Pharmacology/OSPSuite.Core#2198]
At the moment, when the object references in event assignments formulas are resolved, the **reference assignment itself** is used as the reference point (and **not the assigned entity**). 
I think this is wrong and should be changed.

E.g. check this example and try to create a simulation: [Event_Assignment_Path_Resolution.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/files/13963468/Event_Assignment_Path_Resolution.zip)

In my spatial structure, I have 2 containers under _Organism_ : _Organ1_ and _Organ2_. 
Both containers have parameters _P1_ and _P2_.
![grafik](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/assets/25061876/a6d93685-fe40-4068-be09-c51e6a92b4c5)   ![grafik](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/assets/25061876/3fa21270-3543-4a93-83db-4dcae3e24e9f)

I define a new event, which should set _P1 := P2_ in both organs (with _P2_ from the same organ as _P1_).
So I define a formula `P2 `with _P2_ referenced as `..|P2` and use this formula in both event assignments:
<img width="450" src="https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/assets/25061876/1a540182-41f0-4e42-bbf8-1cf7774f9975">   <img width="250" src="https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/assets/25061876/82920f40-cacb-4acb-b716-9ceae4255190">

Now when trying to create a simulation, I get 2 errors below (1 error per event assignment), which shows that the object reference "`..|P2`" is being resolved relative to the event assignment and not relative to `Organism|OrganX|P1`
![grafik](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/assets/25061876/9970bb77-9811-4a53-9670-1adfc77a67b4)

So now I would need to define 2 different formulas - one for each event assignment, where I reference the parameter P1 e.g. with its absolute path (`Organism|Organ1|P1` for the 1st assignment and `Organism|Organ2|P1` for the 2nd).

This is very inconvenient and should be changed (e.g. with the status quo we cannot avoid formulas duplication described in https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2856)

**Comment by PavelBal** [comment]
Is it a use case for a new keyword? :D

I think the current behavior (assignment is the local reference point) follows the logic of the relative path concept within the suite. Changing this behavior would also break many things, I guess. For example, adding the amount of water of an oral application to the total liquid volume in stomach is given by the formula:

`Liquid + Water` with `Water = ..|..|ProtocolSchemaItem|Amount of water` the amount of water in the particular administration. This would not work any more if the local reference point were the changed entity `Organism|Lumen|Stomach|Liquid`.

I see how the solution you proposed can make certain things easier. So maybe we should indeed introduce a new keyword, like `ENTITY` or `CHANGED_ENTITY`?

**Comment by Yuri05** [comment]
> Is it a use case for a new keyword? :D
> 
> I think the current behavior (assignment is the local reference point) follows the logic of the relative path concept within the suite. Changing this behavior would also break many things, I guess. For example, adding the amount of water of an oral application to the total liquid volume in stomach is given by the formula:
> 
> `Liquid + Water` with `Water = ..|..|ProtocolSchemaItem|Amount of water` the amount of water in the particular administration. This would not work any more if the local reference point were the changed entity `Organism|Lumen|Stomach|Liquid`.
> 
> I see how the solution you proposed can make certain things easier. So maybe we should indeed introduce a new keyword, like `ENTITY` or `CHANGED_ENTITY`?

yeah, you are right, these are 2 different use cases and the current use case would not work anymore.
Maybe a new keyword indeed.. need to think more about it.
